pkg/controller/syncset/syncset_controller.go

```diff
@@ -0,0 +1,439 @@
+// found in each SyncSet object
+func (r *ReconcileSyncSet) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the ClusterDeployment instance
+	cd := &hivev1.ClusterDeployment{}
+
+	err := r.Get(context.TODO(), request.NamespacedName, cd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Object not found, return
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request
+		r.logger.WithError(err).Error("error looking up cluster deployment")
+		return reconcile.Result{}, err
+	}
+
+	cdLog := r.logger.WithFields(log.Fields{
+		"clusterDeployment": cd.Name,
+		"namespace":         cd.Namespace,
+	})
+
+	if !cd.Status.Installed {
+		// Cluster isn't installed yet, return
+		cdLog.Info("cluster installation is not complete")
+		return reconcile.Result{}, nil
+	}
+
+	origCD := cd
+	cd = cd.DeepCopy()
+
+	cdLog.Info("reconciling sync sets for cluster deployment")
+
+	// get all sync sets that apply to cd
+	syncSets, err := r.getRelatedSyncSets(cd)
+	if err != nil {
+		cdLog.WithError(err).Error("unable to list related sync sets for cluster deployment")
+		return reconcile.Result{}, err
+	}
+
+	// get all selector sync sets that apply to cd
+	// selectorSyncSets, err := r.getRelatedSelectorSyncSets(cd)
+	// if err != nil {
+	//      cdLog.WithError(err).Error("unable to list related sync sets for cluster deployment")
+	//  	return reconcile.Result{}, err
+	// }
+
+	// get kubeconfig for the cluster
+	secretName := cd.Status.AdminKubeconfigSecret.Name
+	secretData, err := r.loadSecretData(secretName, cd.Namespace, adminKubeConfigKey)
+	if err != nil {
+		cdLog.WithError(err).Error("unable to load admin kubeconfig")
+		return reconcile.Result{}, err
+	}
+	kubeConfig := []byte(secretData)
+
+	for _, syncSet := range syncSets {
+		ssLog := cdLog.WithFields(log.Fields{"syncSet": syncSet.Name})
+		ssLog.Debug("applying sync set")
+
+		syncSetStatus := findSyncSetStatus(syncSet.Name, cd.Status)
+		err = r.applySyncSet(syncSet, kubeConfig, &syncSetStatus, ssLog)
+		if err != nil {
+			ssLog.WithError(err).Error("unable to apply sync set")
+		}
+		cd.Status.SyncSetStatus = appendOrUpdateSyncSetObjectStatus(cd.Status.SyncSetStatus, syncSetStatus)
+	}
+
+	err = r.updateClusterDeploymentStatus(cd, origCD, cdLog)
+	if err != nil {
+		cdLog.WithError(err).Errorf("error updating cluster deployment status")
+		return reconcile.Result{}, err
+	}
+
+	cdLog.Info("done reconciling sync sets for cluster deployment")
+
+	return reconcile.Result{}, nil
+}
+
+// applySyncSet evaluates all resources and patches in a given SyncSet and applies them to the cluster identified by kubeConfig
+func (r *ReconcileSyncSet) applySyncSet(ss hivev1.SyncSet, kubeConfig []byte, syncSetStatus *hivev1.SyncSetObjectStatus, ssLog log.FieldLogger) error {
+	h := r.applierBuilder(kubeConfig, r.logger)
+
+	for _, resource := range ss.Spec.Resources {
+		info, err := h.Info(resource.Raw)
```

Comment:
You need to first call h.Info on all ss.Spec.Resources. If any error occurs, then add the condition and don't try to sync anything.